sdk/typespec/typespec_derive/src/lib.rs

```diff
@@ -0,0 +1,205 @@
+/// By default, the type must also implement `serde::Deserialize`, or the generated code will not compile.
+///
+/// ## Attributes
+///
+/// The following attributes are supported on the struct itself:
+///
+/// ### `#[typespec(crate)]`
+///
+/// The 'crate' attribute specifies an alternate module path, other than the default of `typespec_client_core`, to reference the typespec client crate.
+///
+/// ```rust
+/// # use typespec_derive::Model;
+/// # use serde::Deserialize;
+/// extern crate typespec_client_core as my_typespec;
+///
+/// #[derive(Model, Deserialize)]
+/// #[typespec(crate = "my_typespec")]
+/// struct MyModel {
+///   value: String
+/// }
+/// ```
+///
+/// ### `#[typespec(format)]`
+///
+/// The format attribute specifies the format of the response body. The default is `json`.
+/// If compiling with the `xml` feature, the value `xml` is also supported.
+///
+/// ```rust
+/// # use typespec_derive::Model;
+/// # use serde::Deserialize;
+/// #[derive(Model, Deserialize)]
+/// #[typespec(format = "xml")]
+/// struct MyModel {
+///   value: String
+/// }
+/// ```
+///
+/// **NOTE:** Using formats other than JSON may require enabling additional features in `typespec_client_core`.
+#[proc_macro_derive(Model, attributes(typespec))]
+pub fn model_derive(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
+    let ast: DeriveInput = syn::parse(input).unwrap();
+    let body = generate_body(ast);
+
+    // We wrap the generated code in a const block to give it a unique scope.
+    let gen = quote::quote! {
+        #[doc(hidden)]
+        const _: () = {
+            #body
+        };
+    };
+    gen.into()
+}
+
+fn generate_body(ast: DeriveInput) -> TokenStream {
+    let mut context = Context::new();
+    let (impl_generics, ty_generics, where_clause) = ast.generics.split_for_impl();
+    let name = &ast.ident;
+
+    // Parse attributes
+    let attrs = Attrs::from_attrs(&ast.attrs, &mut context);
+    if !context.errors().is_empty() {
+        // If there were any errors parsing attrs, emit their compile_error! statements and return.
+        let errs = context.errors();
+        return quote::quote! {
+            #(#errs)*
+        };
+    }
+
+    let format = attrs.format.unwrap_or(Format::Json);
+    let deserialize_body = match format {
+        Format::Json => quote::quote! {
+            body.json().await
+        },
+        Format::Xml => quote::quote! {
+            body.xml().await
+        },
+    };
+
+    // If the standard path is used, we need to add 'extern crate', because it's possible the calling code
+    // depends on typespec_client_core transitively, which means it's not in scope by default.
+    // That's not necessary when using a custom path because we assume the user has done that work.
+    let typespec_import = match attrs.typespec_path {
+        Some(path) => quote::quote! {
+            use #path as _typespec;
```

Comment:
Because `typespec` is actually a separate crate, should this maybe be `_typespec_client_core` to disambiguate in the future?

Reply:
Same below and elsewhere as needed.